### PR TITLE
Ensure all hex codes are upper case

### PR DESCRIPTION
**Pull request by lyndonjardine**
… uppercase when their value updates. #22

### CODE REVIEW

**Comment by ajsaraujo** [comment]
Nice! Thanks for your contribution :)

**Comment by ajsaraujo** [comment]
This works neatly, but you could work it out without repetition. You could handle this logic creating a setter, so everytime you assign to `color.hexCode` you ensure it's uppercase. Something like this:

```ts
class Color {
  set hexCode(color: string) {
    this.hexCode = color.toUpperCase();
  }
}
```

This way you handle data consistency next to the model and don't have to tweak presentation code.

**Comment by lyndonjardine** [comment]
Thank you for the suggestion,
I added in the setter as you suggested, but that resulted in a 'duplicate identifier' error since the constructor had the 'public hexCode' property,
So I modified the the constructor of the class and made the hexCode property private (I also had to rename it), and I replaced it with a getter.

File: src/app/models/color.ts [hunk 1]
```diff
@@ -1,12 +1,21 @@
 import { RandomUtils } from '../utils/random.utils';
 
 export class Color {
-  constructor(public hexCode: string = '') {
-    if (!this.hexCode) {
-      this.hexCode = this.generateRandomHex();
+  //renamed to privateHexCode so it doesn't conflict with the setter
```
Comment (ajsaraujo) [inline src/app/models/color.ts]: This comment is not needed. To tell why things have changes is the versioning system's job, not the code's. This comment doesn't make much sense outside the context of this pull request.

File: src/app/models/color.ts [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
+  constructor(private privateHexCode: string = '') {
+    if (!this.privateHexCode) {
+      this.privateHexCode = this.generateRandomHex();
     }
   }
 
+  //used to ensure every hex code is uppercase
```
Comment (ajsaraujo) [inline src/app/models/color.ts]: You can remove this comment as well. It's already clear what the setter is doing.

File: src/app/models/color.ts [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
+  set hexCode(color: string){
+    this.privateHexCode = color.toUpperCase();
+  }
+  get hexCode(){
```
Comment (ajsaraujo) [inline src/app/models/color.ts]: Could you add a line break between these two lines of code?

File: src/app/models/color.ts [hunk 4]
```diff
@@ -1,12 +1,21 @@
 import { RandomUtils } from '../utils/random.utils';
 
 export class Color {
-  constructor(public hexCode: string = '') {
-    if (!this.hexCode) {
-      this.hexCode = this.generateRandomHex();
+  //renamed to privateHexCode so it doesn't conflict with the setter
+  constructor(private privateHexCode: string = '') {
```
Comment (ajsaraujo) [inline src/app/models/color.ts]: Could you rename `privateHexCode` to `_hexCode`? Using underscores is the "standard" in these kinds of situations

**Review by ajsaraujo — CHANGES_REQUESTED** [review]
Just some minor style fixes. Sorry for taking so long.

File: src/app/models/color.ts (hunk 4 above)
Comment (lyndonjardine) [reply src/app/models/color.ts]: Made the change,
It was my original intention to name it that, but it breaks one of the eslint rules
`Unexpected dangling '_' in '_hexCode'.`

So I added `/* eslint-disable no-underscore-dangle */`

Is that acceptable?

Comment (ajsaraujo) [reply src/app/models/color.ts]: Yes, for sure! I don't quite agree with that rule.

**Pull request merged**